AppTaskQueue/test/suites/run-load-test.sh

```diff
@@ -0,0 +1,473 @@
+COMMAND
+
+# Save DSN string and projects config to variables
+PG_DSN="dbname=appscale-test-project user=appscale password=appscale-pwd host=${POSTGRES_VM_PRIVATE_IP}"
+POSTGRES_PROJECT='postgres-test-project'
+CASSANDRA_PROJECT='cassandra-test-project'
+
+log "### Initializing Zookeeper at ${USER}@${ZOOKEEPER_VM} ###"
+ssh -o StrictHostKeyChecking=no -i ${KEY_LOCATION} ${USER}@${ZOOKEEPER_VM} << COMMANDS
+    set -e
+    # Run general zookeeper provisioning script
+    sudo /tmp/prepare-zookeeper.sh
+
+    # Configure project with Cassandra as a backend for Pull Queues
+    sudo /usr/share/zookeeper/bin/zkCli.sh create \
+        /appscale/projects/${CASSANDRA_PROJECT} ""
+
+    # Configure project with Postgres as a backend for Pull Queues
+    sudo /usr/share/zookeeper/bin/zkCli.sh create \
+        /appscale/projects/${POSTGRES_PROJECT} ""
+    sudo /usr/share/zookeeper/bin/zkCli.sh delete \
+        /appscale/projects/${POSTGRES_PROJECT}/postgres_dsn
+    sudo /usr/share/zookeeper/bin/zkCli.sh create \
+        /appscale/projects/${POSTGRES_PROJECT}/postgres_dsn "${PG_DSN}"
+COMMANDS
+
+log "### Initializing Cassandra at ${USER}@${CASSANDRA_VM} ###"
+ssh -o StrictHostKeyChecking=no -i ${KEY_LOCATION} ${USER}@${CASSANDRA_VM}  << CMD
+    set -e
+    sudo /tmp/prepare-cassandra.sh --private-ip ${CASSANDRA_VM_PRIVATE_IP} \
+                                   --zk-ip ${ZOOKEEPER_VM}
+CMD
+
+# Generate comma-separated list of ports from (50000) to (50000 + TQ_PER_VM)
+TQ_PORTS=$(echo $(seq 50000 $((50000 + TQ_PER_VM - 1))))
+# Start all TaskQueue servers
+for tq_vm in ${TASKQUEUE_VMS}
+do
+    log "### Starting ${TQ_PER_VM} Taskqueue servers at ${USER}@${tq_vm} ###"
+    ssh -o StrictHostKeyChecking=no -i ${KEY_LOCATION} ${USER}@${tq_vm} << COMMAND
+        set -e
+        sudo /tmp/restart-taskqueue.sh --ports "${TQ_PORTS// /,}" \
+                                       --db-ip "${CASSANDRA_VM_PRIVATE_IP}" \
+                                       --zk-ip "${ZOOKEEPER_VM}" \
+                                       --lb-ip "${LOADBALANCER_VM}" \
+                                       --source-dir /tmp/AppTaskQueue
+COMMAND
+done
+
+log "### Configuring loadbalancer at ${USER}@${LOADBALANCER_VM} ###"
+ssh -o StrictHostKeyChecking=no -i ${KEY_LOCATION} ${USER}@${LOADBALANCER_VM} << CMD
+    set -e
+    sudo /tmp/configure-loadbalancer.sh \
+        --tq-locations-file /tmp/tq-locations.txt \
+        --listen-tcp ${LOADBALANCER_VM}:4000
+CMD
+
+TQ_LOCATION="${LOADBALANCER_VM}:4000"
+
+
+log ""
+log "=============================================="
+log "=== Preparing Virtualenv for the load test ==="
+log "=============================================="
+# aiohttp lib which is used in e2e test requires Python>=3.5.3
+# test scripts uses new syntax for formatting strings (3.6+)
+PYTHON=$(realpath --strip ./python-3.6.6-plus)
+${HELPERS_DIR}/ensure-python-version.sh 3.6.6 --create-link "${PYTHON}"
```

Comment:
I don't think we should be installing a different version of Python on a user's machine automatically if they just want to run a load test.

I would be more comfortable just exiting with an error message here if their version is not compatible.

Reply:
Agree, it was over-engineering.